agent/scripts/fetch-kernel.sh

```
@@ -0,0 +1,33 @@
+set -euo pipefail
+
+ROOT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")/.." && pwd)"
+LOCK_FILE="${1:-$ROOT_DIR/kernel.lock}"
+DEST_DIR="${KERNEL_DEST_DIR:-$ROOT_DIR/.cache/kernels}"
+
+if [[ ! -f "$LOCK_FILE" ]]; then
+  echo "Lock file not found: $LOCK_FILE" >&2
+  exit 1
+fi
+
+source "$LOCK_FILE"
```

Shellcheck (0.11.0)

[warning] ShellCheck can't follow non-constant source. Use a directive to specify location.

(SC1090)